aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1148,44 +1194,12 @@ Tensor all(const Tensor& self) {
   return _all(result, iter);
 }
 
-Tensor all(const Tensor& self, int64_t dim, bool keepdim) {
-  // Refer [all, any : uint8 compatibility]
-  Tensor result;
-  if (self.scalar_type() == ScalarType::Byte){
-    result = at::empty({0}, self.options());
-  } else {
-    result = at::empty({0}, self.options().dtype(kBool));
-  }
-
-  return at::native::all_out(self, dim, keepdim, result);
-}
-
-Tensor &all_out(const Tensor &self, int64_t dim, bool keepdim, Tensor &result) {
-  TORCH_CHECK(self.device().is_cpu() || self.is_cuda(),
-              "all only supports CPU AND CUDA device type, got: ", self.device().type());
-  TORCH_CHECK(self.layout() == Layout::Strided,
-              "all only supports strided layout, got: ", self.layout());
-  // Refer [all, any : uint8 compatibility]
-  TORCH_CHECK(result.scalar_type() == ScalarType::Bool || result.scalar_type() == ScalarType::Byte,
-              "all only supports bool tensor for result, got: ", result.scalar_type());
-
-  auto out_dtype = result.scalar_type();
-  dim = maybe_wrap_dim(dim, self.dim());
-  if (_dimreduce_return_trivial(result, self, 1, dim, keepdim)) {
-    return result;
-  } else {
-    if (self.is_cuda()) {
-      // As CUDA supports dynamic type casting, we use this overload of
-      // `make_reduction`, which doesn't cast input to the result type i.e. kBool.,
-      // otherwise we use the overload below which casts the input to kBool (which is
-      // an extra operation).
-      auto iter = make_reduction(
-          "all", result, self, dim, keepdim, self.scalar_type(), out_dtype);
-      return _all(result, iter);
-    }
-    auto iter =
-        make_reduction("all", result, self, dim, keepdim, /*out_dtype=*/out_dtype);
-    return _all(result, iter);
+TORCH_IMPL_FUNC(all_out)
+(const Tensor& self, int64_t dim, bool keepdim, const Tensor& result) {
+  auto iter = get_reduction_iter(self, result, dim, keepdim);
```

Comment:
If you can make `all` subclass from TensorIteratorBase, then you won't need this (and you won't risk duplicating the code between the meta and impl methods.